### DISCUSSION TITLE
Proper way to manual mesh level, z-offset and re-do it

### DISCUSSION THREAD

**Discussion by xantrk** [Jyers/Marlin#199] [Q&A] [answered]
Hello,

I am a happy user of the precompiled firmware, but have some questions regarding manual mesh leveling. So I believe similar to normal manual leveling, mesh leveling also requires repeating the process. Up to now, I was using the following procedure:

1) Set z offset to 0
2) Enter leveling menu, level 3x3 points save.
3) Calibrate z offset whether printing live or now, it is typically around 2 babysteps down (-0.02mm I guess).

Now, when I want to re-do the mesh level tomorrow, should I repeat this whole process with canceling z offset first? Without doing this, I didnt manage to succeed as I believe offset sticks while leveling, so every mesh babystep is 2 steps more, than I manually add more offset.

Also, can manual mesh leveling override z-stop, sometimes I hear the z-stop click while I am mesh leveling. Should I adjust my z stop?

**Comment by Jyers** [comment, marked as answer]
For manual mesh leveling, it is probably best to reset the z offset first, as I believe it would affect the axis while you make your mesh. The mesh is able to override the endstop, but it is capped to 1mm below the stop, which is less than the full travel of the switch, making it completely safe.
